### PR TITLE
Remove deprecate warnings

### PR DESCRIPTION
**Pull request by SergioEstevao**
This PR removes the majority of deprecation warning and activates them again on the project configuration.

It only lefts 5 warnings that will need some extra discussion or deeper changes that I will open has different issues:

- PocketAPI SDK - remove all the custom sharing activities #4600
- PrivateSiteURLProtocol - this may need to upgrade all our network code to use NSURLSession.
- WPWebVideoViewController - this class looks i's not being used so I think it will be deleted and removed.

Needs Review: @jleandroperez

### CODE REVIEW

File: WordPress/Classes/Categories/UIActionSheet+Helpers.h [hunk 1]
```diff
@@ -1,19 +0,0 @@
...
-
-
-
-typedef void (^UIActionSheetCompletion) (NSString *buttonTitle);
-
-#pragma mark =====================================================================================
-#pragma mark UIActionSheet Helpers
-#pragma mark =====================================================================================
-
-@interface UIActionSheet (Helpers)
```
Comment (jleandroperez) [inline WordPress/Classes/Categories/UIActionSheet+Helpers.h]: See you, UIActionSheet!

File: WordPress/Classes/Utility/NotificationsManager.m [hunk 2]
```diff
@@ -56,23 +56,15 @@ + (void)registerForPushNotifications
...
-        // iOS 8 or higher notifications registration
-        [[UIApplication sharedApplication] registerForRemoteNotifications];
-
-        // Add the categories to UIUserNotificationSettings
-        UIUserNotificationType types = UIUserNotificationTypeBadge | UIUserNotificationTypeSound | UIUserNotificationTypeAlert;
-        UIUserNotificationSettings *settings = [UIUserNotificationSettings settingsForTypes:types categories:[self buildNotificationCategories]];
-
-        // Finally, register the notification settings
-        [[UIApplication sharedApplication] registerUserNotificationSettings:settings];
-    }
```
Comment (jleandroperez) [inline WordPress/Classes/Utility/NotificationsManager.m]: (Thank you for killing this!). I'm in the process of rewriting this whole class (#4562)

File: WordPress/Classes/Utility/WPError.m [hunk 3]
```diff
@@ -136,7 +137,7 @@ + (void)showAlertWithTitle:(NSString *)title message:(NSString *)message withSup
     [self showAlertWithTitle:title message:message withSupportButton:showSupport okPressedBlock:nil];
 }
 
-+ (void)showAlertWithTitle:(NSString *)title message:(NSString *)message withSupportButton:(BOOL)showSupport okPressedBlock:(void (^)(UIAlertView *))okBlock
```
Comment (jleandroperez) [inline WordPress/Classes/Utility/WPError.m]: (One of these days i'm just gonna kill WPError)

File: WordPress/Classes/ViewRelated/Me/MeViewController.m [hunk 4]
```diff
@@ -370,20 +370,21 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
...
+                                            handler:nil];
+                [alertController addActionWithTitle:NSLocalizedString(@"Disconnect", @"Button for confirming disconnecting WordPress.com account")
+                                              style:UIAlertActionStyleDestructive
+                                            handler:^(UIAlertAction *alertAction) {
+                                                [self signoutFromWordPressAccount];
+                                            }];
+                UITableViewCell *cell = [tableView cellForRowAtIndexPath:indexPath];
+                alertController.popoverPresentationController.sourceRect = [cell bounds];
+                alertController.popoverPresentationController.sourceView = cell;
+                [self presentViewController:alertController animated:YES completion:nil];
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Me/MeViewController.m]: Just heads up. @koke recently rewrote `MeViewController` in Swift. Not sure who is gonna merge first, but his changes should get picked!.

File: WordPress/Classes/ViewRelated/Post/PostSettingsViewController.m [hunk 5]
```diff
@@ -896,14 +895,7 @@ - (void)showMediaPicker
...
     picker.showMostRecentFirst = YES;
-    if (IS_IPAD) {
-        self.popover = [[UIPopoverController alloc] initWithContentViewController:picker];
-        CGRect frame = [self.tableView rectForRowAtIndexPath:[NSIndexPath indexPathForRow:0 inSection:PostSettingsSectionFeaturedImage]];
-        self.popover.delegate = self;
-        [self.popover presentPopoverFromRect:frame inView:self.view permittedArrowDirections:UIPopoverArrowDirectionAny animated:YES];
-    } else {
-        [self presentViewController:picker animated:YES completion:nil];
-    }
+    [self presentViewController:picker animated:YES completion:nil];
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Post/PostSettingsViewController.m]: +1. Much simpler!

File: WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m [hunk 6]
```diff
@@ -123,13 +122,6 @@ - (void)viewWillAppear:(BOOL)animated
 {
     [super viewWillAppear:animated];
 
-    if (self.previousOrientation != UIInterfaceOrientationUnknown) {
-        if (UIInterfaceOrientationIsPortrait(self.previousOrientation) != UIInterfaceOrientationIsPortrait(self.interfaceOrientation)) {
-            [self.tableViewHandler refreshCachedRowHeightsForWidth:CGRectGetWidth(self.view.frame)];
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m]: Ummmmmm i think this change would *definitely* bring side effects. Perhaps we'd just need to switch over...

```
- (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id <UIViewControllerTransitionCoordinator>)coordinator
``` 

cc'ing @aerych!

File: WordPress/Classes/ViewRelated/Settings/ActivityLogDetailViewController.m [hunk 7]
```diff
@@ -53,45 +52,14 @@ - (void)viewWillDisappear:(BOOL)animated
...
-                                                        otherButtonTitles:NSLocalizedString(@"Mail", @""), nil];
-
-        [actionSheet showFromBarButtonItem:sender animated:YES];
-    }
-
+    UIActivityViewController *activityViewController = [[UIActivityViewController alloc] initWithActivityItems:@[self.logText]
+                                                                                         applicationActivities:nil];
+    activityViewController.modalPresentationStyle = UIModalPresentationPopover;
+    activityViewController.popoverPresentationController.barButtonItem = sender;
+    [self presentViewController:activityViewController animated:YES completion:nil];
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Settings/ActivityLogDetailViewController.m]: My god. This is a beautiful touch sir!

Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Settings/ActivityLogDetailViewController.m]: This was (probably) already there. But... the popover is misaligned with regards of the navBar button.

[See screenshot here](https://cloudup.com/cEvI6_DFwuV)

**Comment by jleandroperez** [comment]
@SergioEstevao thanks for taking care of this one!. Only one critical comment added (re: `ReaderCommentsViewController`).

Perhaps we should wait until the remaining issues are solved, before re-enabling the deprecation compiler flag?

File: WordPress/Classes/ViewRelated/Me/MeViewController.m (hunk 4 above)
Comment (koke) [inline WordPress/Classes/ViewRelated/Me/MeViewController.m]: First! https://github.com/wordpress-mobile/WordPress-iOS/pull/4589#event-497487482

**Comment by koke** [comment]
I merged develop in 6097a28 and fixed the conflict with MeViewController myself.

File: WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m (hunk 6 above)
Comment (aerych) [inline WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m]: Nice catch @jleandroperez.  
Yeah the idea here was the user might rotate the device while viewing a link or video.  When they return, layout would need to be refreshed.

Comment (SergioEstevao) [inline WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m]: @jleandroperez I reverted those changes @aerych will handle this in a separate PR.

**Comment by SergioEstevao** [comment]
@jleandroperez I reverted the change that could give us some issues.

At the moment they are only 8 left so I don't think they get on the way. So I'm planning to merge it with the warnings on, because this will serve us as reminder to fix them and avoid us to add more of these kind of warnings going on .

File: WordPress/Classes/ViewRelated/Me/MeViewController.m [hunk 8]
```diff
@@ -0,0 +1,474 @@
+// Me contents:
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Me/MeViewController.m]: @SergioEstevao this one sounds like a merge glitch perhaps?

Comment (SergioEstevao) [inline WordPress/Classes/ViewRelated/Me/MeViewController.m]: This must happen when @koke merge is changes on this branch.

Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Me/MeViewController.m]: This one should go go (because of its Swift reimplementation!)

Comment (SergioEstevao) [inline WordPress/Classes/ViewRelated/Me/MeViewController.m]: Killed the file @jleandroperez, ready to merge?

**Comment by jleandroperez** [comment]
@SergioEstevao sounds good!. 

:shipit:

**Pull request merged**